scvi/model/_scvi.py

```diff
@@ -188,4 +203,72 @@ def setup_anndata(
             fields=anndata_fields, setup_method_args=setup_method_args
         )
         adata_manager.register_fields(adata, **kwargs)
+        # register new fields for latent mode if needed
+        latent_mode = _get_latent_adata_type(adata)
+        if latent_mode is not None:
+            adata_manager.register_new_fields(cls._get_latent_fields(latent_mode))
         cls.register_manager(adata_manager)
+
+    def _get_latent_adata_from_adata(
+        self,
+        mode: LatentDataType,
+        use_latent_key: str = "X_latent",
+        use_latent_qzm_key: str = "X_latent_qzm",
+        use_latent_qzv_key: str = "X_latent_qzv",
+    ) -> AnnData:
+        all_zeros = csr_matrix(self.adata.X.shape, dtype=np.int8).toarray()
```

Comment:
```suggestion
        all_zeros = csr_matrix(self.adata.X.shape, dtype=np.float32)
```
 toarray will create a dense array?